apps/cowswap-frontend/src/modules/trade/hooks/useAmountsToSign.ts

```diff
@@ -0,0 +1,29 @@
+import { useMemo } from 'react'
+
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+
+import { useDerivedTradeState } from './useDerivedTradeState'
+import { useReceiveAmountInfo } from './useReceiveAmountInfo'
+
+export interface AmountsToSign {
+  maximumSendSellAmount: CurrencyAmount<Currency>
+  minimumReceiveBuyAmount: CurrencyAmount<Currency>
+}
+
+/**
+ * Returns amounts that will end up in signed order
+ * It means, those values already include fees and slippage depending on order type
+ */
+export function useAmountsToSign(): AmountsToSign | null {
+  const { supportBuyOrders, inputCurrencyAmount, outputCurrencyAmount } = useDerivedTradeState() || {}
+  const { afterSlippage } = useReceiveAmountInfo() || {}
+
+  return useMemo(() => {
+    const maximumSendSellAmount = supportBuyOrders ? afterSlippage?.sellAmount : inputCurrencyAmount
+    const minimumReceiveBuyAmount = supportBuyOrders ? afterSlippage?.buyAmount : outputCurrencyAmount
```

Comment:
The setting name is a bit confusing because it's not only about buy orders support, is it?
I say that because the fact that it's a buy order impacts only the sell amount, no?

Reply:
https://github.com/cowprotocol/cowswap/pull/5484#discussion_r1985018529